docs/07-lifecycle.md

```diff
@@ -2,8 +2,68 @@
 ## Development
 ## Submit to Marketplace
 ## Provision
+### App Bundle
+Bundle is a convenient way to deliver an app to the Mattermost ecosystem. It provides a way to organize code and resources needed for an app to run. A bundle is created by the developer of the app. Mattermost uses bundles to provision and install/uninstall apps. 
+App bundle contains `manifest.json` file, `static/` folder and one or several lambda function bundles.
+- `static/` folder contains all the static files app needs. For the Mattermost AWS Apps static files are automatically provisioned and stored in the dedicated AWS S3 bucket(`mattermost-apps-bucket-production`). Apps can use them whenever they wish by providing the static file name to the Proxy Plugin. For the third-party hosted AWS apps static files are stored in the different S3 bucket(specified by the third-party). For the HTTP Apps, when creating a server, developer should store the static files in the `/static/file_name` relative URL.
```

Comment:
I'm 3/5 against using the term `Proxy Plugin` as it isn't obvious what the term means. Also, the plugin is more the a proxy. Can we use something more obvious as `Apps Plugin`?